index.cgi

```diff
@@ -34,18 +34,25 @@ sign() {
       ip) ip=$varvalue ;;
       ns) ns=$varvalue ;;
       days) days=$varvalue ;;
+      gp) gp=$varvalue ;;
+      dn) dn=$varvalue ;;
     esac
   done
 
-  [ -z "$cn" ] && die "cn= is mandatory"
+  [ -z "$cn" -a -z "$dn" ] && die "Either cn= or dn= is mandatory"
+  [ -n "$cn" -a -n "$dn" ] && die "Pick either cn or dn"
+  [ -z "$dn" ] && dn="/CN=$cn$( for i in ${gp//,/ }; do echo -n "/O=$i"; done)"
```

Comment:
What about leave cn as simple as `/CN=<cn>`? The user can send a dn if he needs another fields.